src/completers/completeState/index.js

```diff
@@ -0,0 +1,22 @@
+import isStringArray from '../validate';
+
+// Given a defaultState, it populates that state with ${key}Loading and ${key}Error
+function completeState(defaultState, ignoredTargets = []) {
+  if (!defaultState || defaultState.constructor !== Object) {
+    throw new Error('Expected an object as a state to complete.');
+  }
+  if (!isStringArray(ignoredTargets)) {
+    throw new Error('Expected an array of strings as ignored targets');
+  }
+
+  const completedState = { ...defaultState };
+  Object.keys(defaultState)
+    .filter(key => ignoredTargets.indexOf(key) === -1)
```

Comment:
using indexOf here may be buggy. why not `===` ?

Reply:
I'm not getting this point :stuck_out_tongue: